python/pipeline/stack.py

```diff
@@ -977,6 +978,129 @@ def make(self, key):
         self.insert1({**key, 'resized': resized, 'lcned': lcned, 'sharpened': sharpened})
 
 
+@schema
+class StackSurfaceMethod(dj.Lookup):
+    definition = """ # Methods used to compute surface of the brain
+
+    surface_method_id   : tinyint unsigned   # Unique ID given to each surface calculation method
+    ---
+    method_title        : varchar(32)        # Title of surface calculation method
+    method_description  : varchar(256)       # Details on surface calculation
+    """
+
+    contents = [
+        [1, 'Paraboloid Fit', 'Fit ax^2 + by^2 + cx + dy + f to surface after finding max of sobel']
+    ]
+
+
+@schema
+class StackSurface(dj.Computed):
+    definition = """ # Calculated surface of the brain
+
+    -> PreprocessedStack
+    -> StackSurfaceMethod
+    ---
+    surface_im        : longblob          # Matrix of calculated depth for each pixel in stack
+    lower_bound_im    : longblob          # Lower bound of 95th percentile confidence interval
+    upper_bound_im    : longblob          # Upper bound of 95th percentile confidence interval
+    """
+
+    def make(self, key):
+
+        # WARNINGS
+        #  - This code assumes the surface will be in the top half of the stack
+        #      - Only the top half of z-values are analyzed
+        #  - Points along the edge are dropped to avoid errors due to blank space left by stack registration
+        #  - This code assumes the surface median intensity should be in the bottom 60% of the range of values over z
+        #      - ex. Intensities ranges from 10-20. Surface points must have an intensity < .6*(20-10) + 10 = 17.5
+        #      - This is within the 2r x 2r window being analyzed
+        #  - This code drops any 2r x 2r field where the first median value is above the 30th-percentile of the whole stack.
+        #  - Windows where the final median intensity is below 10 are removed
+        #      - Attempts to replace this with a percentile all fail
+        #  - This code drops guessed depths > 95th-percentile and < 5th-percentile to be more robust to outliers
+
+        def surface_eqn(data, a, b, c, d, f):
+            x, y = data
+            return a * x ** 2 + b * y ** 2 + c * x + d * y + f
+
+        # SETTINGS
+        # Note: Intial parameters for fitting set further down
+        r = 50  # Radius of square in pixels
```

Comment:
do you think these parameters will be changed at some point, maybe we could have a part table here called SurfaceParams with these params, that way if they ever change we will know what surfaces where computed each way. Although I guess if we are gonna change the params maybe we should just create a new SurfaceMethod altogether, up to you.

Reply:
Talking with Jake, I think we're going to make a new method if the settings need to be changed.